src/App.js

```diff
@@ -89,14 +99,48 @@ function App() {
     setCardsText([...newCards]);
   };
 
+  const addToDeleteListHandler = (cardId, isSelected, isEdited) => {
+    //adding selected cards to the list
+    if (isSelected && !isEdited) {
+      setDeleteList((prevList) => [...prevList, cardId]);
+    //deleting cards from the list if they were unselected
+    } else if ((!isSelected && !isEdited) || (isEdited && !isSelected)) {
```

Comment:
Per previously added logic there is rule 'If record goes to Edit mode, checkbox becomes unchecked.'. So this condition is excess.